src/extract/gadm.py

```diff
@@ -1,13 +1,24 @@
 import os
+import zipfile
 
-from utils.yaml_api import parse_config
+from utils.yaml_api import parse_yaml
 from utils.requests_api import download_url
 
+import pdb
 
-GADM_YEMEN_ADDRESS = 'https://biogeo.ucdavis.edu/data/gadm3.6/gpkg/gadm36_YEM_gpkg.zip'
+
+def get_world():
```

Comment:
This method should take the URL and output filename as input variables. See e.g. `extract/cod.py`

Reply:
In `extract/cod.py` it looks like there are two functions. Can see that the snakemake rule `extract_adm0_cod` references it's own params which it references from `config.yml`.  
It then references the entry_point (as defined in `setup.py`) `get_adm0_snakemake`. 
In `extract/cod.py` the function get_adm0_snakemake then simply references the function in the same file get_adm0 passing it the params it got for the snakefile. 
Not sure why get_adm0_snakemake is necessary given the config file params could just be called directly from get_adm0 unless I'm missing something a principle regarding snakemake?

Reply:
You're right that `get_adm0_snakemake` is not strictly necessary. The reason I created a separate method is because I wanted to keep a clean version of the method with proper variable names as input arguments. That way the method is independent of snakemake.

Reply:
👍